resources/Microsoft.VSCode.Dsc/Microsoft.VSCode.Dsc.psm1

```diff
@@ -5,44 +5,77 @@ $ErrorActionPreference = 'Stop'
+        'X86' { return $Insiders ? @('{C26E74D1-022E-4238-8B9D-1E7564A36CC9}_is1', '{26F4A15E-E392-4887-8C09-7BC55712FD5B}_is1') : @('{D628A17A-9713-46BF-8D57-E671B46A741E}_is1', '{F8A2A208-72B3-4D61-95FC-8A65D340689B}_is1') }
+        'Arm' { Throw 'Not supported.' }
+        'Arm64' { return $Insiders ? @('{69BD8F7B-65EB-4C6F-A14E-44CFA83712C0}_is1', '{0AEDB616-9614-463B-97D7-119DD86CCA64}_is1') : @('{D9E514E7-1A56-452D-9337-2990C0DC4310}_is1', '{A5270FC5-65AD-483E-AC30-2C276B63D0AC}_is1') }
+        Default { Throw 'Could not determine architecture.' }
+    }
+}
 
-        $insidersMachineInstallLocation = TryGetRegistryValue -Key "$($machineUninstallRegistry)\$($vsCodeInsidersMachineProductCode)" -Property $installLocationProperty
-        if ($insidersMachineInstallLocation) {
-            return $insidersMachineInstallLocation + $cmdPath
-        }
-    } else {
-        $cmdPath = 'bin\code.cmd'
-        $codeUserInstallLocation = TryGetRegistryValue -Key "$($userUninstallRegistry)\$($vsCodeUserProductCode)" -Property $installLocationProperty
-        if ($codeUserInstallLocation) {
-            return $codeUserInstallLocation + $cmdPath
+function Get-VSCodeCLIPath {
+    param (
+        [switch]$Insiders
+    )
+
+    # Get the available keys
+    $registryKeys = Get-OSArchitectureRegistryKey -Insiders:$Insiders.IsPresent
+    $registryHive = @('HKCU:\Software\Microsoft\Windows\CurrentVersion\Uninstall', 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall', 'HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall')
+
+    if ($IsLinux) {
+        if ($Insiders) {
+            $InstallLocation = Join-Path ($env:PATH.Split([System.IO.Path]::PathSeparator) -match 'Microsoft VS Code Insiders') 'code-insiders'
+            if (Test-Path $InstallLocation -ErrorAction SilentlyContinue) {
+                return $InstallLocation
+
+            }
+        } else {
+            $InstallLocation = Join-Path ($env:PATH.Split([System.IO.Path]::PathSeparator) -match 'Microsoft VS Code') 'code'
+            if (Test-Path $InstallLocation -ErrorAction SilentlyContinue) {
+                return $InstallLocation
+            }
         }
+    }
 
-        $codeMachineInstallLocation = TryGetRegistryValue -Key "$($machineUninstallRegistry)\$($vsCodeMachineProductCode)" -Property $installLocationProperty
-        if ($codeMachineInstallLocation) {
-            return $codeMachineInstallLocation + $cmdPath
+    if ($IsWindows) {
+        foreach ($hive in $registryHive) {
+            foreach ($key in $registryKeys) {
+                Write-Verbose -Message ("Searching path '{0}' with key '{1}'" -f $hive, $key)
+                $installLocation = TryGetRegistryValue -Key "$hive\$key" -Property 'InstallLocation'
+                if ($installLocation) {
+                    $cmdPath = $Insiders ? 'bin\code-insiders.cmd' : 'bin\code.cmd'
+                    return Join-Path $installLocation $cmdPath
+                }
+            }
```

Comment:
I'm not a fan of iterating through registry key entries until we find a match. Based on the parameters provided (scope, isInsiders) for the DSC resource, it should be deterministic as to where the VSCode should be installed.

Reply:
Nor am I buddy. The thing is, DSC doesn't know the scope; WinGet does. We don't have a property for it. It is only 3 locations it iterates through, and if it finds one, it returns it. What do you think? Can we push this through, or do you have another suggestion for it?